pkg/traces/config.go

```diff
@@ -235,7 +248,13 @@ func exporter(rwCfg RemoteWriteConfig) (map[string]interface{}, error) {
 		return nil, errors.New("must have a configured a backend endpoint")
 	}
 
-	headers := map[string]string{}
+	// TODO: investigate
+	// For some reason, jaeger exporter factory makes this field nil when otlp's makes it a 0len map
```

Comment:
So, it turns out that the `otlpexporter` sets the headers to an empty map in the [default config](https://github.com/open-telemetry/opentelemetry-collector/blob/v0.41.0/exporter/otlpexporter/factory.go#L49). This is what's causing the diff between otlp and jaeger. Can we document this briefly?

Also, I think it'd be more correct to change the code to
```go
if rwCfg.Format == formatOtel {
```